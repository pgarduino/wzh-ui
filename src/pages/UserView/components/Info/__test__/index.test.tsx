import { render } from '@testing-library/react';

import { UserViewInfo } from '..';

describe("UserViewInfo", () => {
    beforeAll(() => {
        Object.defineProperty(window, "matchMedia", {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
                matches: false,
                media: query,
                onchange: null,
                addListener: jest.fn(), // Deprecated
                removeListener: jest.fn(), // Deprecated
                addEventListener: jest.fn(),
                removeEventListener: jest.fn(),
                dispatchEvent: jest.fn(),
            }))
        });
    });

    const user = {
        "id": 10,
        "name": "Clementina DuBuque",
        "username": "Moriah.Stanton",
        "email": "Rey.Padberg@karina.biz",
        "address": {
            "street": "Kattie Turnpike",
            "suite": "Suite 198",
            "city": "Lebsackbury",
            "zipcode": "31428-2261",
            "geo": {
                "lat": "-38.2386",
                "lng": "57.2232"
            }
        },
        "phone": "024-648-3804",
        "website": "ambrose.net",
        "company": {
            "name": "Hoeger LLC",
            "catchPhrase": "Centralized empowering task-force",
            "bs": "target end-to-end models"
        }
    }
    test('Should render', async () => {
        const wrapper = render(
                <UserViewInfo user={user} />
        )
        expect(wrapper).toMatchSnapshot()
    });
})