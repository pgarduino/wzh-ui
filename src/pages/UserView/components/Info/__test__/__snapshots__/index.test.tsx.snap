// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserViewInfo Should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="ant-descriptions"
        >
          <div
            class="ant-descriptions-header"
          >
            <div
              class="ant-descriptions-title"
            >
              User Info
            </div>
          </div>
          <div
            class="ant-descriptions-view"
          >
            <table>
              <tbody>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="1"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        Id
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        10
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-descriptions-item"
                    colspan="1"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        Name
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        Clementina DuBuque
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-descriptions-item"
                    colspan="1"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        UserName
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        Moriah.Stanton
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="1"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        Email
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        Rey.Padberg@karina.biz
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-descriptions-item"
                    colspan="1"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        Address
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        street: 
                        Kattie Turnpike
                        <br />
                        city: 
                        Lebsackbury
                        <br />
                        zipcode: 
                        31428-2261
                        <br />
                        street: 
                        Kattie Turnpike
                        <br />
                        geo: 
                        -38.2386
                         
                        57.2232
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-descriptions-item"
                    colspan="1"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        Phone
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        024-648-3804
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="1"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        Website
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        ambrose.net
                      </span>
                    </div>
                  </td>
                  <td
                    class="ant-descriptions-item"
                    colspan="2"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        Company
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                      >
                        name: 
                        Hoeger LLC
                        <br />
                        catchPhrase: 
                        Centralized empowering task-force
                        <br />
                        bs: 
                        target end-to-end models
                      </span>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="ant-descriptions"
      >
        <div
          class="ant-descriptions-header"
        >
          <div
            class="ant-descriptions-title"
          >
            User Info
          </div>
        </div>
        <div
          class="ant-descriptions-view"
        >
          <table>
            <tbody>
              <tr
                class="ant-descriptions-row"
              >
                <td
                  class="ant-descriptions-item"
                  colspan="1"
                >
                  <div
                    class="ant-descriptions-item-container"
                  >
                    <span
                      class="ant-descriptions-item-label"
                    >
                      Id
                    </span>
                    <span
                      class="ant-descriptions-item-content"
                    >
                      10
                    </span>
                  </div>
                </td>
                <td
                  class="ant-descriptions-item"
                  colspan="1"
                >
                  <div
                    class="ant-descriptions-item-container"
                  >
                    <span
                      class="ant-descriptions-item-label"
                    >
                      Name
                    </span>
                    <span
                      class="ant-descriptions-item-content"
                    >
                      Clementina DuBuque
                    </span>
                  </div>
                </td>
                <td
                  class="ant-descriptions-item"
                  colspan="1"
                >
                  <div
                    class="ant-descriptions-item-container"
                  >
                    <span
                      class="ant-descriptions-item-label"
                    >
                      UserName
                    </span>
                    <span
                      class="ant-descriptions-item-content"
                    >
                      Moriah.Stanton
                    </span>
                  </div>
                </td>
              </tr>
              <tr
                class="ant-descriptions-row"
              >
                <td
                  class="ant-descriptions-item"
                  colspan="1"
                >
                  <div
                    class="ant-descriptions-item-container"
                  >
                    <span
                      class="ant-descriptions-item-label"
                    >
                      Email
                    </span>
                    <span
                      class="ant-descriptions-item-content"
                    >
                      Rey.Padberg@karina.biz
                    </span>
                  </div>
                </td>
                <td
                  class="ant-descriptions-item"
                  colspan="1"
                >
                  <div
                    class="ant-descriptions-item-container"
                  >
                    <span
                      class="ant-descriptions-item-label"
                    >
                      Address
                    </span>
                    <span
                      class="ant-descriptions-item-content"
                    >
                      street: 
                      Kattie Turnpike
                      <br />
                      city: 
                      Lebsackbury
                      <br />
                      zipcode: 
                      31428-2261
                      <br />
                      street: 
                      Kattie Turnpike
                      <br />
                      geo: 
                      -38.2386
                       
                      57.2232
                    </span>
                  </div>
                </td>
                <td
                  class="ant-descriptions-item"
                  colspan="1"
                >
                  <div
                    class="ant-descriptions-item-container"
                  >
                    <span
                      class="ant-descriptions-item-label"
                    >
                      Phone
                    </span>
                    <span
                      class="ant-descriptions-item-content"
                    >
                      024-648-3804
                    </span>
                  </div>
                </td>
              </tr>
              <tr
                class="ant-descriptions-row"
              >
                <td
                  class="ant-descriptions-item"
                  colspan="1"
                >
                  <div
                    class="ant-descriptions-item-container"
                  >
                    <span
                      class="ant-descriptions-item-label"
                    >
                      Website
                    </span>
                    <span
                      class="ant-descriptions-item-content"
                    >
                      ambrose.net
                    </span>
                  </div>
                </td>
                <td
                  class="ant-descriptions-item"
                  colspan="2"
                >
                  <div
                    class="ant-descriptions-item-container"
                  >
                    <span
                      class="ant-descriptions-item-label"
                    >
                      Company
                    </span>
                    <span
                      class="ant-descriptions-item-content"
                    >
                      name: 
                      Hoeger LLC
                      <br />
                      catchPhrase: 
                      Centralized empowering task-force
                      <br />
                      bs: 
                      target end-to-end models
                    </span>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
